src/ClientImpl.cpp

```diff
@@ -345,8 +357,8 @@ void ClientImpl::initiateAuthentication(PendingConnection* pc, struct buffereven
         throw voltdb::LibEventException();
     }
     AuthenticationRequest authRequest( m_username, "database", m_passwordHash );
-    ScopedByteBuffer bb(authRequest.getSerializedSize());
-    authRequest.serializeTo(&bb);
+    ScopedByteBuffer bb(authRequest.getSerializedSize(m_hashScheme));
```

Comment:
Why doesn't the AuthenticationRequest take the hash scheme in its constructor? It's weird to pass it to the next two lines.